Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m

```diff
@@ -125,6 +125,7 @@ - (NSData *)transportBytes {
   // Encode a 2nd time to actually get the bytes from it.
   size_t bufferSize = sizestream.bytes_written;
   CFMutableDataRef dataRef = CFDataCreateMutable(CFAllocatorGetDefault(), bufferSize);
+  CFDataSetLength(dataRef, bufferSize);
```

Comment:
Remove the second call, below?

Reply:
Yeah, not done yet. I just wrote this in github to remind myself to do something about it.

Reply:
Done, and unit test added. Confirmed that without these changes the unit test fails. Good catch!